### ISSUE TITLE
Addons no longer working (e.g. Offline Mode)?

### ISSUE THREAD

**Issue by Joel889** [Xyrio/RSSOwl#5]
Thank you for picking up this project!

Are addons no longer supported?
I was using a 6KB offline addon, which adds the option File > Work Offline, to limit RSSOwl to the cached article feed. Using Wireshark, I noticed that RSSOwl makes an online request every time one would click on an article. This was true even for text only articles, presumably to check whether the article has been updated since downloading the feed.

This behavior is also documented here:
http://tutorial.rssowl.org/ht_offline.html

"When online, RSSOwl reloads news summaries and retains them in memory *each time* you select a newsfeed."

Anyway, is it possible to run addons? 
I tried putting it into "p2" and also into the "plugins" folder. Or can an offline mode be implemented?

P.S.: Your sourceforge forum post regarding placing java inside the installation folder without having to install it is a great tip! Perhaps adding this to the main page would be helpful to many users.

**Comment by Xyrio** [comment]
on the release page i try to explain that you have to go to the menu and manually select the unpacked folder:
under Help / Install New Software... add a new site by pointing to the unzipped folder containing the site.xml for example like: file:/C:\update_site_addons

**Comment by Joel889** [comment]
Where do I get the site.xml file?
All I have is the addon (.jar file), and I was able to find feature.xml from the feature folder. I tried pointing it to the jar file, feature.xml file, and also renaming the feature.xml to site.xml.

Feature.xml has the following content:

<?xml version="1.0" encoding="UTF-8"?>
<feature
      id="org.rssowl.feature.contrib.offline"
      label="Offline Mode"
      version="1.0.0"
      provider-name="RSSOwl.org">

   <description>
      This add-on will add a new entry to the File menu to switch to an offline modus where feeds will no longer update or attachments downloaded.
   </description>

   <url>
      <update label="RSSOwl.org" url="http://boreal.rssowl.org/update/addons/"/>
   </url>

   <requires>
      <import plugin="org.eclipse.ui"/>
      <import plugin="org.eclipse.core.runtime"/>
      <import plugin="org.rssowl.ui" version="2.1.2" match="greaterOrEqual"/>
      <import plugin="org.rssowl.core"/>
   </requires>

   <plugin
         id="org.rssowl.contrib.offline"
         download-size="5"
         install-size="0"
         version="1.0.0.201108111128"
         unpack="false"/>

</feature>

**Comment by Xyrio** [comment]
on the release page where u downloaded 2.4.0 is also a download for the updated addons including the offline mode one: https://github.com/Xyrio/RSSOwl/releases
the mentioned update_site_addons_release_2017_12_10.zip file contains what you need.

**Comment by Joel889** [comment]
Thank you! I got it to work, but noticed the following:
1) The addon installation would fail unless the .rssowl2 profile folder was on drive F:. 
I had changed the profile folder's relative location to be stored in the same folder as the main program, and all feed items were always available regardless of the drive's drive letter.

2) After installing the addon, a new config.ini was created in folder "config240" (the old version is still in the "configuration" folder), but the profile path in the new config.ini was restored back to default.

**Comment by Xyrio** [comment]
yes you need to change the configuration\config.ini so that it uses the .rssowl2 folder from where you want. the difference to 2.2.1 is that plugins are now installed in the .rssowl2-beta (or however you changed it) folder. i intentionally changed folders in case there are bugs. config221 is not used. old .rssowl2 folder is now .rssowl2-beta as mentioned. unless you change it in configuration\config.ini

**Comment by Joel889** [comment]
I guess what I'm saying is that I changed "configuration/config.ini" when I first started using **your** version and everything was working. But as soon as I installed an addon, all my settings and feeds were (seemingly) lost, because it created another profile somewhere else. I then had to go "config240/config.ini" and point it back to my old profile.

So it seems that if no addons are installed, "configuration/config.ini" is being used. As soon as addons are installed "config240/config.ini" is being used.

**Comment by Xyrio** [comment]
yea i tried it and the very first time you install an addon it writes the default paths to the .rssowl2-beta\config240\config.ini you need to correct those before you restart. afterwards it doesnt seem to do it anymore. even when uninstalling all addons an then installing a first addon again it wont do it.

**Comment by Joel889** [comment]
I tried version 2.6, but still cannot get "Offline Mode" to work in a custom user directory.
The initial installation works, but when changing the location of the configuration folder, "offline mode" located in the menu bar disappears. Reinstalling fails, because it says that the addon is already installed.

"Offline Mode" + custom configuration is working fine in the original version 2.21.

**Comment by Xyrio** [comment]
there is a second config.ini inside the home folder: \.rssowlnix-beta\config\config.ini
when the addon is (re)installed the config.ini inside the custom homefolder is reset. if you change the 2 path entries again it should work.

**Comment by Joel889** [comment]
I tried that several times but it doesn't work.

I did get it working in a version prior to 2.6.

To recap, version < 2.6 can only do one of the following (but not both at the same time):
a) Set a custom profile folder 
b) Run plugins.

**Comment by Xyrio** [comment]
i am using 2.7.0 with modified path with the notes addon myself so i know it works.

so here are the steps:
- modify 2 lines in `RSSOwlnix\configuration\config.ini`
    1. `osgi.instance.area.default=@user.home/.rssowlnix-beta-bla`
    2. `osgi.configuration.area=@user.home/.rssowlnix-beta-bla/config`
- now rssowl uses the modified path
- install addon(s) (installs it to the current path which is the modified path)
- after restart it goes to wrong path
- modify 2 lines in `.rssowlnix-beta-bla\config` 
- now rssowl uses again modified path

needs this fix only once, somehow it remembers now (until a new version?)

**Comment by Joel889** [comment]
Thank you. I followed your steps and got it working again. I wish there was an easier way to save the profile folder within the application for portable applications.
